types/index.d.ts

```diff
@@ -173,6 +186,7 @@ declare namespace StyleDictionary {
     registerTemplate(this: Core, template: Named<{ template: string }>): this;
     registerAction(this: Core, action: Named<Action>): this;
     registerFilter(this: Core, filter: Filter): this;
+    registerParser(this: Core, parser: Parser): this;
```

Comment:
Incidentally... On further testing, I'm not sure exactly how `registerParser` is supposed to work. The types look ok. But when I initialised a style dictionary as follows:

```ts
const myDictionary = require('style-dictionary').extend(myConfigObject)
myDictionary.registerParser(myParser)
myDictionary.buildAllPlatforms()
```

Also this didn't work:
```ts
const myDictionary = require('style-dictionary')
myDictionary.registerParser(myParser)
myDictionary.extend(myConfigObject)
myDictionary.buildAllPlatforms()
```

The my customer parsers _are not_ executed. The only way I could get this to work was to put my custom parser inside `myConfigObject`, via: ` parsers: [myParser]`. 

It looks like the `extend` function calls `combineJSON` immediately (i.e. before we've run `registerParser`), rather than waiting for `buildAllPlatforms` to run, hence `registerParser` is called too late. I don't suppose you have any examples how I'm suppose to run this?

Reply:
Can you confirm you are on `3.0.0-rc4` version? I know there was an issue with parsers not being run on a previous release candidate version... 

I just tried it out and these both work:

```javascript
// build.js
const StyleDictionary = require('style-dictionary');
const yaml = require('yaml');

const config = {
  // .. config
};

StyleDictionary.registerParser({
  pattern: /\.yaml$/,
  parse: ({contents, filePath}) => yaml.parse(contents)
});

StyleDictionary.extend(config).buildAllPlatforms();
```

Where the script to build would be `node build.js`

```javascript
// sd.config.js
const StyleDictionary = require('style-dictionary');
const yaml = require('yaml');

const config = {
  // .. config
};

StyleDictionary.registerParser({
  pattern: /\.yaml$/,
  parse: ({contents, filePath}) => yaml.parse(contents)
});

module.exports = config;
```

Where the script to run would be `style-dictionary build --config ./sd.config.js`

You are correct that you would need to register the parsers __before__ you called `.extend()`. This is specific for parsers, where registering a transform for example could happen after `.extend()` but would have to happen before `.buildAllPlatforms()`